src/main/java/Akshan.java

```diff
@@ -0,0 +1,179 @@
+            processMarkUnmark(splitInput[0], taskList, false);
+            break;
+        case TODO:
+        case DEADLINE:
+        case EVENT:
+            processTask(commandType, command, taskList);
+            break;
+        default:
+            System.out.println("Uh oh, no command found in: " + command);
+        }
+    }
+
+    /**
+     * Processes a task command (TODO, DEADLINE, or EVENT).
+     *
+     * @param taskType The type of task.
+     * @param command The full command string.
+     * @param taskList The list of tasks.
+     */
+    private static void processTask(CommandType taskType, String command, TaskList taskList) {
+        String[] params = COMMAND_PARSERS.get(taskType).apply(command);
+        Task task = Task.createTask(taskType.getCommand(), params);
+        addTaskToList(taskList, task);
+    }
+
+    /**
+     * Adds a task to the task list and prints a confirmation message.
+     *
+     * @param taskList The list of tasks.
+     * @param task The task to be added.
+     */
+    private static void addTaskToList(TaskList taskList, Task task) {
+        taskList.addItem(task);
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + task);
+        System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+    }
+
+    /**
+     * Processes the mark or unmark command.
+     *
+     * @param command The command string.
+     * @param taskList The list of tasks.
+     * @param mark True if marking, false if unmarking.
+     */
+    private static void processMarkUnmark(String command, TaskList taskList, boolean mark) {
+        String[] parts = command.split(" ");
+        if (parts.length == 2) {
+            int index = Integer.parseInt(parts[1]);
+            taskList.setItemStatus(index, mark);
+        }
+    }
+
+    /**
+     * Parses a TODO command.
+     *
+     * @param command The full command string.
+     * @return An array containing the task description.
+     */
+    private static String[] parseTodo(String command) {
+        return new String[]{command.split(" ", 2)[1]};
+    }
+
+    /**
+     * Parses a DEADLINE command.
+     *
+     * @param command The full command string.
+     * @return An array containing the task description and deadline.
+     */
+    private static String[] parseDeadline(String command) {
+        String[] parts = command.split(" /by ", 2);
+        return new String[]{parts[0].split(" ", 2)[1], parts[1]};
+    }
+
+    /**
+     * Parses an EVENT command.
+     *
+     * @param command The full command string.
+     * @return An array containing the event description, start time, and end time.
+     */
+    private static String[] parseEvent(String command) {
+        String[] parts = command.split(" /", 3);
+        return new String[]{
+                parts[0].split(" ", 2)[1],
+                parts[1].split(" ", 2)[1],
+                parts[2].split(" ", 2)[1]
+        };
+    }
```

Comment:
I like the abstraction of parseEvent! Overall your methods work well, but it might be good to add some error handling for cases where the input isn't formatted quite right. For instance, checking if inputs are null or if commands are in the expected format.

Reply:
Thanks for the advice! Will take note on the point of adding validation!